### PR TITLE
Make the Web Locks `uniqueName` helper (likely) globally unique

### LINKED ISSUE #15015: web-locks/signal.tentative.https.any.serviceworker.html is flaky
https://github.com/web-platform-tests/wpt/pull/14420 revealed that web-locks/signal.tentative.https.any.serviceworker.html is flaky. This can be seen by running `./wpt run --verify --channel dev --binary `which google-chrome-unstable` --log-tbpl - chrome /web-locks/signal.tentative.https.any.serviceworker.html` locally, producing this table:

|                  Subtest                   |          Results           |                Messages               |
|--------------------------------------------|----------------------------|---------------------------------------|
|                                            | OK                         |                                       |
| `The signal option must be an AbortSignal` | PASS                       |                                       |
| `Passing an already aborted signal aborts` | PASS                       |                                       |
| `An aborted request results in AbortError` | **FAIL: 5/10, PASS: 5/10** | `assert_equals: expected 1 but got 2` |
| `Abort after a timeout`                    | **FAIL: 4/10, PASS: 6/10** | `assert_equals: expected 1 but got 2` |
| `Signal that is not aborted`               | PASS                       |                                       |
| `Synchronously signaled abort`             | PASS                       |                                       |
| `Abort signaled after lock granted`        | PASS                       |                                       |
| `Abort signaled after lock released`       | PASS                       |                                       |

These are the flaky subtests:

https://github.com/web-platform-tests/wpt/blob/7d538d126b49411702cdd88354f520cdf9a65702/web-locks/signal.tentative.https.any.js#L38-L63

https://github.com/web-platform-tests/wpt/blob/7d538d126b49411702cdd88354f520cdf9a65702/web-locks/signal.tentative.https.any.js#L65-L94

The test came from https://github.com/web-platform-tests/wpt/pull/12449 / https://github.com/web-platform-tests/wpt/pull/12860 by @inexorabletash and reviewed by @pwnall.

### PR DESCRIPTION
**Pull request by foolip**
Fixes https://github.com/web-platform-tests/wpt/issues/15015.

### CODE REVIEW

**Comment by Ms2ger** [comment]
I think we'd usually use `token()` from `common/utils.js` to get a unique identifier; I haven't checked if it'd be appropriate here.

**Comment by inexorabletash** [comment]
Using `token()` sgtm

No "realistic" constraints on lock names.

File: web-locks/resources/helpers.js [hunk 1]
```diff
@@ -4,9 +4,9 @@
   // Generate a unique resource identifier, using the script path and
   // test case name. This is useful to avoid lock interference between
   // test cases.
-  let res_num = 0;
   self.uniqueName = (testCase, prefix) => {
-    return `${self.location.pathname}-${prefix}-${testCase.name}-${++res_num}`;
+    const rand = Math.floor(1e10 * Math.random());
```
Comment (pwnall) [inline web-locks/resources/helpers.js]: If it helps, I've used this in some personal projects -- `Math.random().toString(36).substr(2)`

**Review by inexorabletash — APPROVED** [review]
Ooops, forgot to actually review this.

This is fine, but using `token()` would be better, wouldn't it?

**Comment by foolip** [comment]
Yes, I should use `token()`, I just neglected this PR. Only trouble is that all tests that use `uniqueName()` will have to know that it depends on `token()` and also include common/utils.js. That's not very elegant.

**Comment by foolip** [comment]
Closing this before it's two years old, since clearly I'm not going to prioritize fixing this using `token()` correctly.

**Pull request closed without merging**